sdk/dotnet/Pulumi/Resources/InvokeArgs.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright 2016-2019, Pulumi Corporation
+
+using System;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// Base type for all invoke argument classes.
+    /// </summary>
+    public abstract class InvokeArgs : InputArgs
+    {
+        public static readonly InvokeArgs Empty = new EmptyInvokeArgs();
+
+        protected InvokeArgs()
+        {
+        }
+
+        private protected override void ValidateMember(Type memberType, string fullName)
+        {
+            if (typeof(IInput).IsAssignableFrom(memberType))
+            {
+                throw new InvalidOperationException($"{fullName} must not be an Input<T>");
+            }
```

Comment:
almost hte same as ResourceArgs except that with Invoke args you absolutely *cannot* use `Input<T>`.